corehq/apps/case_search/utils.py

```diff
@@ -119,3 +122,73 @@ def _add_case_property_queries(self):
                 self.search_es = self.search_es.xpath_query(self.domain, query)
             else:
                 self.search_es = self.search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+
+
+def get_related_cases(domain, app_id, case_type, cases):
+    """
+    Fetch related cases that are necessary to display any related-case
+    properties in the app requesting this case search.
+
+    Returns list of CommCareCase objects for adding to CaseDBFixture.
+    """
+    if not cases:
+        return []
+
+    app = get_app_cached(domain, app_id)
+    paths = get_related_case_relationships(app, case_type)
+    if not paths:
+        return []
+    return get_related_case_results(domain, cases, paths)
+
+
+def get_related_case_relationships(app, case_type):
+    """
+    Get unique case relationships used by search details in any modules that
+    match the given case type and are configured for case search.
+    It will miss related cases that are referenced only by calculated properties
+
+    Returns a set of paths, e.g. {"parent", "host", "parent/parent"}
+    """
+    paths = set()
+    for module in app.get_modules():
+        if module.case_type == case_type and module_offers_search(module):
+            for column in module.search_detail("short").columns + module.search_detail("long").columns:
+                if not column.useXpathExpression:
+                    parts = column.field.split("/")
+                    if len(parts) > 1:
+                        parts.pop()     # keep only the relationship: "parent", "parent/parent", etc.
+                        paths.add("/".join(parts))
+    return paths
+
+
+def get_related_case_results(domain, cases, paths):
+    """
+    Given a set of cases and a set of case property paths,
+    fetches ES documents for all cases referenced by those paths.
+    """
+    if not cases:
+        return []
+
+    results_cache = {}
+    for path in paths:
+        current_cases = cases
+        parts = path.split("/")
+        for index, identifier in enumerate(parts):
+            fragment = "/".join(parts[:index + 1])
+            if not fragment:
+                continue
```

Comment:
I think this condition will never be true - it'd only be hit if `parts` is empty (and `path == ""`), but in that case, the body of this `for` loop wouldn't execute.

Reply:
Thanks, that's from an earlier iteration where I was passing in paths that included the property. Removed.